src/lib/dnssd/minimal_mdns/responders/Responder.h

```diff
@@ -34,6 +35,38 @@ class ResponderDelegate
     virtual void AddResponse(const ResourceRecord & record) = 0;
 };
 
+/// Controls specific options for responding to mDNS queries
+///
+class ResponseConfiguration
+{
+public:
+    ResponseConfiguration() {}
+    ~ResponseConfiguration() = default;
+
+    chip::Optional<uint32_t> GetTtlSecondsOverride() const { return mTtlSecondsOverride; }
+    ResponseConfiguration & SetTtlSecondsOverride(chip::Optional<uint32_t> override)
+    {
+        mTtlSecondsOverride = override;
+        return *this;
+    }
+
+    ResponseConfiguration & SetTtlSecondsOverride(uint32_t value) { return SetTtlSecondsOverride(chip::MakeOptional(value)); }
```

Comment:
Just curious, is there some max values that we should reject?

Reply:
I believe any number seems valid.
I imagine we would want more code to have some way to return validity, maybe some IsValid on the config. The API right now is builder-styled.

For now I think we can live with this as is. This is internal API so misuse is less likely (we only use it for 0).